model/common/tests/common/interpolation/mpi_tests/test_parallel_interpolation.py

```diff
@@ -0,0 +1,207 @@
+from icon4py.model.common.decomposition import definitions as decomposition
+from icon4py.model.common.grid import horizontal as h_grid
+from icon4py.model.common.interpolation import (
+    interpolation_attributes as attrs,
+    interpolation_factory,
+)
+from icon4py.model.testing import definitions as test_defs, parallel_helpers, test_utils
+
+from ...fixtures import (
+    backend,
+    data_provider,
+    decomposition_info,
+    download_ser_data,
+    experiment,
+    geometry_from_savepoint,
+    grid_savepoint,
+    icon_grid,
+    interpolation_factory_from_savepoint,
+    interpolation_savepoint,
+    processor_props,
+    ranked_data_path,
+)
+from ..unit_tests.test_interpolation_factory import assert_reordered
+
+
+if TYPE_CHECKING:
+    import gt4py.next.typing as gtx_typing
+
+    from icon4py.model.testing import serialbox as sb
+
+
+vert_lb_domain = h_grid.vertex_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_2)
+
+
+@pytest.mark.level("integration")
+@pytest.mark.datatest
+@pytest.mark.mpi
+@pytest.mark.parametrize("processor_props", [True], indirect=True)
+@pytest.mark.parametrize(
+    "attrs_name, intrp_name, atol",
+    [
+        (attrs.C_LIN_E, "c_lin_e", 1e-12),
+        (attrs.NUDGECOEFFS_E, "nudgecoeff_e", 1e-12),
+        (attrs.C_BLN_AVG, "c_bln_avg", 1e-11),
+        (attrs.E_FLX_AVG, "e_flx_avg", 1e-10),
+        (attrs.E_BLN_C_S, "e_bln_c_s", 1e-12),
+        (attrs.GEOFAC_DIV, "geofac_div", 1e-12),
+        (attrs.GEOFAC_N2S, "geofac_n2s", 1e-12),
+        (attrs.POS_ON_TPLANE_E_X, "pos_on_tplane_e_x", 1e-12),
+        (attrs.POS_ON_TPLANE_E_Y, "pos_on_tplane_e_y", 1e-8),
+        (attrs.CELL_AW_VERTS, "c_intp", 1e-12),
+    ],
+)
+def test_distributed_interpolation_attrs(
+    backend: gtx_typing.Backend,
+    interpolation_savepoint: sb.InterpolationSavepoint,
+    grid_savepoint: sb.IconGridSavepoint,
+    experiment: test_defs.Experiment,
+    processor_props: decomposition.ProcessProperties,
+    decomposition_info: decomposition.DecompositionInfo,
+    interpolation_factory_from_savepoint: interpolation_factory.InterpolationFieldsFactory,
+    attrs_name: str,
+    intrp_name: str,
+    atol: int,
+) -> None:
+    parallel_helpers.check_comm_size(processor_props)
+    intp_factory = interpolation_factory_from_savepoint
+    field_ref = interpolation_savepoint.__getattribute__(intrp_name)()
+    field_ref = field_ref.asnumpy()
+    field = intp_factory.get(attrs_name).asnumpy()
+    assert test_utils.dallclose(
+        field, field_ref, atol=atol, rtol=5e-9
+    ), f"comparison of {attrs_name} failed"
+
+
+@pytest.mark.level("integration")
+@pytest.mark.datatest
+@pytest.mark.mpi
+@pytest.mark.parametrize("processor_props", [True], indirect=True)
+@pytest.mark.parametrize(
+    "attrs_name, intrp_name, index",
+    [
+        (attrs.GEOFAC_GRG_X, "geofac_grg", 0),
+        (attrs.GEOFAC_GRG_Y, "geofac_grg", 1),
+    ],
+)
+def test_distributed_interpolation_grg(
```

Comment:
If you separate out the test then I would not split the x, y components. They are computed together anyway, you can simply assert them both in the same test case, that saves one computation because the factories are setup from scratch in each test function (as the fixtures have default scope)